### PR TITLE
refactor(config): superblocks and map

### PR DESCRIPTION
**Pull request by moT01**
This PR refactors the superblock-order stuff in preparation for https://github.com/freeCodeCamp/freeCodeCamp/issues/50412. It should greatly simplify things for us. The superblock order will now be the same across all i18n's, and the maps will look (mostly) the same on all the pages.

Since German doesn't have the new RWD translated, this removes it from the menu until that is translated.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: config/superblocks.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  ]
+};
+
+Object.freeze(notAuditedSuperBlocks);
+
+type Config = {
+  language?: string;
+  showNewCurriculum: string | undefined;
+  showUpcomingChanges: string | undefined;
+};
```
Comment (Sboonny) [inline config/superblocks.ts:174]:
```suggestion
type Config = {
  showNewCurriculum: string | undefined;
  showUpcomingChanges: string | undefined;
};

type LanguageConfig = Config & {
    language: Languages;
}
```

File: config/superblocks.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+  language,
+  showNewCurriculum,
+  showUpcomingChanges
+}: Config): SuperBlocks | null {
+  const flatSuperBlockMap = createFlatSuperBlockMap({
+    showNewCurriculum,
+    showUpcomingChanges
+  });
+  for (const superBlock of flatSuperBlockMap) {
+    if (notAuditedSuperBlocks[language as Languages].includes(superBlock)) {
```
Comment (Sboonny) [inline config/superblocks.ts:217]:
```suggestion
    if (notAuditedSuperBlocks[language].includes(superBlock)) {
```

File: config/superblocks.ts [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    throw Error(`'${language}' key not found in 'notAuditedSuperBlocks'`);
+  }
+
+  const flatSuperBlockMap = createFlatSuperBlockMap({
+    showNewCurriculum,
+    showUpcomingChanges
+  });
+  const auditedSuperBlocks = flatSuperBlockMap.filter(
+    superBlock =>
+      !notAuditedSuperBlocks[language as Languages].includes(superBlock)
```
Comment (Sboonny) [inline config/superblocks.ts:239]:
```suggestion
      !notAuditedSuperBlocks[language].includes(superBlock)
```

**Review by Sboonny — COMMENTED** [review]
small nitpick, because it is using the wrong type

File: config/superblocks.ts [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+  }
+  return Object.values(superBlockMap).flat();
+}
+
+// this is so we know where to display the "help us translate" section
+export function getFirstNotAuditedSuperBlock({
+  language,
+  showNewCurriculum,
+  showUpcomingChanges
+}: Config): SuperBlocks | null {
```
Comment (Sboonny) [inline config/superblocks.ts:211]:
```suggestion
}: LanguageConfig): SuperBlocks | null {
```

File: config/superblocks.ts [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    }
+  }
+  return null;
+}
+
+export function getAuditedSuperBlocks({
+  language = 'english',
+  showNewCurriculum = 'false',
+  showUpcomingChanges = 'false'
+}: Config): SuperBlocks[] {
```
Comment (Sboonny) [inline config/superblocks.ts:228]:
```suggestion
}: LanguageConfig): SuperBlocks[] {
```

**Comment by Sboonny** [comment]
@moT01 is it okay if I push a commit that sort, I have suggested part of it by mistake.

**Comment by moT01** [comment]
Do it

**Review by Sboonny — APPROVED** [review]
I will need to refactor enum to objects to the one that doesn't change and maps to the one that does change, to get the true type of the values and we won't need to use `as` to tell TypeScript about the true type.

Unfortunately, this is outside the scope of this PR, and should be addressed in issue first, if similar situation arise.

LGTM

**Comment by ojeytonwilliams** [comment]
I'll give this a proper review soon, but one quick question: do we need to change the infrastructure at all? i.e. are we just removing German from the UI or are we taking the servers down?

**Comment by moT01** [comment]
Just removing from the UI until they finish the new RWD.

**Review by Sembauke — APPROVED** [review]

**Pull request merged**